### PR TITLE
Closes #353| Create dataset loader for FilWordNet

### LINKED ISSUE #353: Create dataset loader for FilWordNet
Dataloader name: `filwordnet/filwordnet.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?filwordnet

|     Dataset| filwordnet |
|-------------|---|
| Description | Filipino WordNet (FilWordNet) is a lexical database of Filipino language. It was derived from the Princeton WordNet and translated by humans to Filipino. It documents 13,539 unique words and 9,519 synsets. Each synset includes the definition, part-of-speech, word senses, and Suggested Upper Merged Ontology terms (SUMO terms). |
| Subsets | - |
| Languages | fil |
| Tasks | Wordnet |
| License | Unknown (unknown) |
| Homepage | https://github.com/danjohnvelasco/Filipino-WordNet |
| HF URL | - |
| Paper URL | https://www.researchgate.net/publication/228938676_Introducing_Filipino_WordNet |

### PR DESCRIPTION
**Pull request by danjohnvelasco**
Closes #353 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [ ] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

> Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.

This dataset is a lexicon. It does not support any tasks. This case is similar to https://github.com/SEACrowd/seacrowd-datahub/pull/155 and https://github.com/SEACrowd/seacrowd-datahub/pull/332

To run tests:
```
python -m tests.test_seacrowd_source_only seacrowd/sea_datasets/filwordnet/filwordnet.py
```

### CODE REVIEW

File: seacrowd/sea_datasets/filwordnet/filwordnet.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+_SUPPORTED_TASKS = []
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class FilWordNet(datasets.GeneratorBasedBuilder):
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/filwordnet/filwordnet.py:63]:
```suggestion
class FilWordNetDataset(datasets.GeneratorBasedBuilder):
```

File: seacrowd/sea_datasets/filwordnet/filwordnet.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        )
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            # TODO: Create your source schema here
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/filwordnet/filwordnet.py:83]:
```suggestion
```

File: seacrowd/sea_datasets/filwordnet/filwordnet.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            for row in csv_reader:
+                if is_first_row:  # skip first row, they are column names
+                    is_first_row = False
+                    continue
+
+                rows.append(row)
+
+        if self.config.schema == "source":
+            for key, row in enumerate(rows):
+                example = {"word_id": row[0], "lemma": row[1], "synset_id": row[2], "sense_id": row[3], "pos": row[4], "lexdomain_id": row[5], "definition": row[6], "last_modifier": row[7], "sumo": row[8]}
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/filwordnet/filwordnet.py:136]: nit: This line may be too long. One technique I do is to add a comma at the end (i.e., at the end of `"sumo": row[8]`) and let the formatter arrange it in a readable format.

**Review by ljvmiranda921 — COMMENTED** [review]
Dataloader works properly, thank you @danjohnvelasco !

I assume the reason why there's no SEACrowd config is because there's no schema that accomodates this dataset, no? 

I have a few nits but thank you for this PR!

**Comment by SamuelCahyawijaya** [comment]
@ljvmiranda921  @danjohnvelasco : 
> I assume the reason why there's no SEACrowd config is because there's no schema that accomodates this dataset, no?

Sorry for the confusion, I added the `source-only` label to the issue, so we just need to implement the source schema for this dataset.

**Review by SamuelCahyawijaya — APPROVED** [review]
Hi @danjohnvelasco , thank you for contributing! The data loader works perfectly.
There are a few nits from @ljvmiranda921, please update it accordingly and afterward we can merge the PR.

**Comment by danjohnvelasco** [comment]
Hi @ljvmiranda921, I updated the code according to your suggestion. Thanks!

**Pull request merged**